src/config/job-config/actions/urlaction/urlaction.ts

```diff
@@ -21,16 +21,22 @@ export class URLJobAction<T extends JobDto> implements JobAction<T> {
     const url = encodeURI(this.urlTemplate(job));
     Logger.log(`(Job ${job.id}) Requesting ${url}`, "URLAction");
 
+    const headers = this.headerTemplates
+      ? Object.fromEntries(
+          Object.entries(this.headerTemplates).map(([key, template]) => [
+            key,
+            template(job, {
+              data: {
+                authToken: this.authorizationToken,
+              },
+            }),
```

Comment:
```suggestion
            template(job, {
              data: {
                env: process.env,
              },
            }),
```

If I understand correctly, I think what @sbliven is suggesting above is allowing the handlebar templates to access the env variables like so or similarly.
This way we can avoid defining `"authTokenEnvVar": "ETHZ_ARCHIVER_BASIC_AUTH_TOKEN"` in jobConfig and directly access the token here:
```
    "headers": {
      "accept": "application/json",
      "content-type": "application/json",
      "Authorization": "Basic {{env.ETHZ_ARCHIVER_BASIC_AUTH_TOKEN}}"
    },
```
What I also like with this approach is that we don't need to pass an extra parameter to the url action constructor, which is currently passed outside of the defined `URLJobActionOptions`.

Reply:
Yes exactly